tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -127,7 +129,14 @@ public void onSuccess(RawResponse response) {
             @Override
             public void onFailure(Throwable throwable) {
                 queuedRequests.decrementAndGet();
-                // TODO handle the failure case
+
+                // TODO better interface for sending errors
+                ErrorMessage error = new ErrorMessage(
+                        rawRequest.getId(),
+                        ErrorType.BadRequest,
```

Comment:
Any exception happens at decoding should be considered as BadRequest (assuming decoding code logic is correct).
Anyway, the current implementation is much better than eating the error. So, I would consider the change a step forward.